components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -168,14 +181,27 @@ export default function ProjectSettingsView() {
                 />
                 {enablePrebuilds && (
                     <>
-                        <InputField label="Workspace machine type" disabled={!enablePrebuilds}>
-                            <div className="max-w-md">
-                                <SelectWorkspaceClassComponent
-                                    disabled={!enablePrebuilds}
-                                    selectedWorkspaceClass={project.settings?.workspaceClasses?.prebuild}
-                                    onSelectionChange={setWorkspaceClassForPrebuild}
-                                />
-                            </div>
+                        <SelectInputField
+                            disabled={!enablePrebuilds}
+                            label="Build branches"
+                            value={prebuildBranchStrategy}
```

Comment:
There ends up being a delay between clicking the prebuilds checkbox and it responding. I believe it's because we're relying on the call to complete and the data to get updated before it's reflected in the UI. I think we should aim to have interaction reflected immediately for most UIs, so when you click a checkbox, it should show checked or unchecked immediately. The same goes for selecting a value from the dropdown, it doesn't change right away, which makes it feel broken.

I think a better pattern for this type of interaction is to encapsulate the update in a react-query mutation, which then lets us optimistically update the value in cache so it's reflected in the UI right away. Example of doing that with workspace descriptions found [here](https://github.com/gitpod-io/gitpod/blob/2faa1b61b79550ec29ae7b4ce35ff9324226084c/components/dashboard/src/data/workspaces/update-workspace-description-mutation.ts#L27-L28).

Reply:
Going to address this. Thanks for the hint @selfcontained!

Reply:
I've pushed a small fix for this issue in https://github.com/gitpod-io/gitpod/pull/18716/commits/31a14091f05d1a46b878b563f1ff0f6f9d691dcd

That's not the recommended mutation change, which I'm plan to do in a follow-up addressing UX improvements, but it solves the mentioned problem.